packages/core/src/components/ui/ProductGallery/ProductGallery.tsx

```diff
@@ -91,8 +91,8 @@ function ProductGallery({
   useProductsPrefetch(prev ? prev.cursor : null)
   useProductsPrefetch(next ? next.cursor : null)
 
-  const hasFacets =
-    Boolean(data?.search?.facets) && data.search.facets.length > 0
+  const hasFacets = Boolean(data?.search?.facets)
```

Comment:
Why remove `data.search.facets.length > 0`?

Reply:
that's because `Skeleton` logic only need to double-check the loading state, and the loading state checks `data?.search?.facets` existence. `data.search.facets.length > 0` means that is not in the loading state, it means the API already returned the facets, but there is no facets in the return, then we don't need to show the filter.